nixos/modules/services/web-apps/privatebin.nix

```
@@ -0,0 +1,278 @@
+      '';
+      example = literalExpression ''
+        {
+          "opcache.interned_strings_buffer" = "8";
+          "opcache.max_accelerated_files" = "10000";
+          "opcache.memory_consumption" = "128";
+          "opcache.revalidate_freq" = "15";
+          "opcache.fast_shutdown" = "1";
+        }
+      '';
+    };
+
+    nginx = mkOption {
+      type = types.submodule (import ../web-servers/nginx/vhost-options.nix { inherit config lib; });
+      default = {};
+      example = literalExpression ''
+        {
+          forceSSL = true;
+          enableACME = true;
+        }
+      '';
+      description = lib.mdDoc ''
+        Optional settings to pass to the nginx virtualHost.
+      '';
+    };
+  };
+
+  config = mkIf cfg.enable {
+    users.users.${user} = {
+      inherit group;
+
+      isSystemUser = true;
+      createHome = false;
+    };
+    users.users.${config.services.nginx.user} = mkIf config.services.nginx.enable { extraGroups = [ group ]; };
+    users.groups.${group} = { };
+
+    services.phpfpm.pools.${user} = {
+      inherit user group;
+      settings = {
+        "listen.mode" = "0660";
+        "listen.owner" = user;
+        "listen.group" = group;
+        "catch_workers_output" = true;
+        "clear_env" = false;
+      } // cfg.poolConfig;
+      phpEnv.CONFIG_PATH = "${configFile}";
+    };
+
+    services.postgresql = mkIf (autoDb == "pgsql") {
+      enable = true;
+      ensureUsers = [{
+        name = user;
+        ensurePermissions = {
+          "DATABASE \"${user}\"" = "ALL PRIVILEGES";
+        };
+      }];
+      ensureDatabases = [
+        user
+      ];
+    };
+
+    services.mysql = mkIf (autoDb == "mysql") {
+      enable = true;
+      package = mkDefault pkgs.mariadb;
+      ensureUsers = [{
+        name = user;
+        ensurePermissions = {
+          "${user}.*" = "ALL PRIVILEGES";
+        };
+      }];
+      ensureDatabases = [
+        user
+      ];
+    };
+    systemd.services."phpfpm-${user}".serviceConfig = {
+      ProtectSystem = "full";
+      PrivateTmp = true;
+      EnvironmentFile = cfg.environmentFiles;
+    };
+    systemd.tmpfiles.rules = flatten [
+      "d /var/lib/privatebin/data 0750 ${user} ${group} - -"
+    ];
+
+    services.nginx = {
+      enable = mkDefault true;
+      virtualHosts.privatebin = mkMerge [
+        cfg.nginx
+        {
+          root = mkForce "${pkgs.privatebin}/share/privatebin";
+          extraConfig = optionalString (cfg.nginx.addSSL || cfg.nginx.forceSSL || cfg.nginx.onlySSL || cfg.nginx.enableACME) "fastcgi_param HTTPS on;";
+          locations = {
+            "/" = {
+              index = "index.php";
+            };
+            "~ \.php$" = {
+              extraConfig = ''
+                try_files $uri $uri/ /index.php?$query_string;
+                include ${pkgs.nginx}/conf/fastcgi_params;
+                fastcgi_param SCRIPT_FILENAME $document_root$fastcgi_script_name;
```

```suggestion
                include ${pkgs.nginx}/conf/fastcgi.conf;
```
Only difference between the two fastcgi configs is this param